homework/index.js

```diff
@@ -29,16 +32,69 @@
     });
     return elem;
   }
+  function app(repositories) {
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Although this logic works well and fulfils all requirements, I think there is still some room for improvement. Ideally, we should write functions in such a way that it is responsible for only one piece of action. For example, the already available functions `createAndAppend` and `fetchJSON` are such functions which are doing only one piece of work.

However, the `app` function here is doing multiple things. It is responsible for:
1. Creating the app layout by calling `createAndAppend` for different tags.
2. Defining logic for putting text into the layout created
3. Fetching contributors information
4. Displaying the fetched contributors information.

I think we can split all these operations in their own functions(which you are already doing) and they need not be contained within the `app` function. Thus, I would suggest that you give it a try and see if you can move them out of `app` function.

For example, `renderContributors` and `innerText` can be moved out of `app` function and if they are using any variable from defined inside `app`, then you can pass those variables as function parameters to them. `innerText` uses `des`, `fork`, `up`, `list2` and `repositories` variables which can be set as function parameters when we move out `innerText` out of the `app` function.

Drop me a message on Slack if you have any questions around it 🙂